### ISSUE TITLE
Should not be able to set private setters

### ISSUE THREAD

**Issue by afscrome** [afscrome/IronVelocity#5]
Attempting to set a property with a private setter currently succeeds.  It should fail, logging an error.  May also affect internal / protected setters.

Trying to call set on a getter only property should be ignored.  It currently throws an exception like the below:
````velocity
#set($myObj.Example = "test")
````
````csharp
public class MyObj
{
    public string Example {get; private set;}
}
````

**Comment by afscrome** [comment]
Looks like the current approach only checks the visibility of the property, not the visibility of the setter.  Related to #4.

**Comment by afscrome** [comment]
Fixed by #30
